pkg/api/kptfile/v1alpha2/types.go

```diff
@@ -0,0 +1,213 @@
+	Pipeline *Pipeline `yaml:"pipeline,omitempty"`
+
+	// Inventory contains parameters for the inventory object used in apply.
+	Inventory *Inventory `yaml:"inventory,omitempty"`
+}
+
+// OriginType defines the type of origin for a package.
+type OriginType string
+
+const (
+	// GitOrigin specifies a package as having been cloned from a git repository.
+	GitOrigin OriginType = "git"
+)
+
+// UpdateStrategyType defines the strategy for updating a package from upstream.
+type UpdateStrategyType string
+
+const (
+	// ResourceMerge performs a structural schema-aware comparison and
+	// merges the changes into the local package.
+	ResourceMerge UpdateStrategyType = "resource-merge"
+	// FastForward fails without updating if the local package was modified
+	// since it was fetched.
+	FastForward UpdateStrategyType = "fast-forward"
+	// ForceDeleteReplace wipes all local changes to the package.
+	ForceDeleteReplace UpdateStrategyType = "force-delete-replace"
+)
+
+// Upstream is a user-specified upstream locator for a package.
+type Upstream struct {
+	// Type is the type of origin.
+	Type OriginType `yaml:"type,omitempty"`
+
+	// Git is the locator for a package stored on Git.
+	Git *Git `yaml:"git,omitempty"`
+
+	// UpdateStrategy declares how a package will be updated from upstream.
+	UpdateStrategy UpdateStrategyType `yaml:"updateStrategy,omitempty"`
+}
+
+// UpstreamLock is a resolved locator for the last fetch of the package.
+type UpstreamLock struct {
+	// Type is the type of origin.
+	Type OriginType `yaml:"type,omitempty"`
+
+	// Git is the locator for a package stored on Git.
+	Git *Git `yaml:"git,omitempty"`
+}
+
+// Git contains information on the origin of packages fetched from a git repository.
+type Git struct {
+	// Repo is the git repository the package was cloned from.  e.g. https://
+	Repo string `yaml:"repo,omitempty"`
+
+	// Directory is the sub directory of the git repository that the package was cloned from
+	Directory string `yaml:"directory,omitempty"`
+
+	// Ref can be a Git branch, tag, or commit ID.
+	Ref string `yaml:"ref,omitempty"`
+
+	// Commit is the git commit that the package was fetched at
+	Commit string `yaml:"commit,omitempty"`
+}
+
+// PackageInfo contains information such as license, documentation, etc.
+// These fields are not consumed by any functionality in kpt and are simply passed through.
+type PackageInfo struct {
+	// URL is the location of the package.  e.g. https://github.com/example/com
+	URL string `yaml:"url,omitempty"`
+
+	// Email is the email of the package maintainer
+	Email string `yaml:"email,omitempty"`
+
+	// License is the package license
+	License string `yaml:"license,omitempty"`
+
+	// Version is a logical package version
+	Version string `yaml:"version,omitempty"`
+
+	// Tags enables humans and tools to attach arbitrary package metadata.
+	Tags []string `yaml:"tags,omitempty"`
+
+	// Man is the path to documentation about the package.
+	Man string `yaml:"man,omitempty"`
```

Comment:
Unless there is strong reason to base this on `man`, I'd recommend calling this `documentation` or `doc`.
the word `man` is also ambiguous, depending on context (i.e. person)